logdb/logdb.go

```diff
@@ -72,8 +74,75 @@ func New(path string) (logDB *LogDB, err error) {
 }
 
 // NewMem create a log db in ram.
+// Unlike the file-based database, this implementation:
+// 1. Uses synchronous=off for faster writes in memory
+// 2. Uses a single connection with minimal transaction overhead
+// 3. Optimizes for in-memory performance
 func NewMem() (*LogDB, error) {
-	return New("file::memory:")
+	// Generate a unique database name to prevent concurrent access issues
+	dbName := fmt.Sprintf("file:memdb_%d?mode=memory&cache=shared&_txlock=immediate&synchronous=off&journal_mode=memory", time.Now().UnixNano())
```

Comment:
nit; In an extreme concurrent scenario, nano timestamp can also be the same, how about random bytes in 4 or 6?

Reply:
Good point - fixed 👍